src/init.cpp

```diff
@@ -1904,3 +1901,9 @@ bool AppInitMain(NodeContext& node, interfaces::BlockAndHeaderTipInfo* tip_info)
 
     return true;
 }
+
+bool StartIndexes(NodeContext& node)
```

Comment:
commit 68b13a2b6c96fd061a31add58b8626c5bf1cdf3d:
Considering how large this function becomes and how low-level the code is in the final state, I don't really love having it in `init.cpp` instead of somewhere in `index/`. Could it be better in `index/base.cpp` as a free function, even if it's not part of `BaseIndex`?

Reply:
I'd definitely avoid moving `StartIndexes` into the `interfaces::Chain` class, because because that class is supposed to expose limited node functionality to outside callers like wallets and indexes without giving them access to node internals. It is supposed to be a thin wrapper around node functions and data, not to contain any real functionality itself. Its methods are meant to be called by wallets and indexes, not by node code because it would add an unnecessary layer of indirection, bring in a monolithic dependency, and could lead to `interfaces::Chain` class growing larger than it needs to be.

I think `StartIndexes` function is actually pretty at home in `src/init.cpp` and would be fine to leave there. But if you want to move it, you could move it to a `src/index/init.cpp` file. If you do move it to `src/index/init.cpp`, it would make index code mirror structure of wallet code. In wallet code, `libbitcoin_wallet.a` library contains all the '.cpp` files in `src/wallet/` **except** `src/wallet/init.cpp`, which is part of `libbitcoin_node.a`. I'm not sure I Iove that structure, but it works and there is precedent for it.

Another options would be to move it somewhere like `src/node/indexes.cpp`. Conceivably the `src/wallet/init.cpp` file could move to `src/node/wallets.cpp` to be consistent with that.

I don't think it matters too much what source file `StartIndexes` function lives in though, as long as it is not part of the chain interface.

Reply:
ok gotcha about the chain interface.

I thought about it because of the goal of removing the node internal types from the subdirectories. If we stick to that path, the code move into the `src/index/init.cpp` could be seen as a regression.

The good point about moving this code somewhere else is that it could be unit tested without having to add `init.h` and `NodeContext` dependencies.

At least for now, I'm a bit inclined for the `src/node/` option but not feeling strong over it neither. I would also be fine if we keep it at `init.cpp` until we add more unit tests.
What do you think @mzumsande?

Reply:
I think I also like the `src/node` option most, but it's ok for me to keep in init for now and move that later.